### PR TITLE
Auto-generate APIRule and Istio CRD docs

### PR DESCRIPTION
**Pull request by nataliasitko**
**Description**

Changes proposed in this pull request:

- update descriptions in APIRule and Istio CRDs
- generate tables from CRDs in Istio and APIRule technical reference docs
- update the table generation Makefile

**Related issue(s)**
- https://github.com/kyma-project/kyma/issues/17480

### CODE REVIEW

File: installation/resources/crds/api-gateway/apirules.gateway.crd.yaml [hunk 1]
```diff
@@ -47,26 +47,27 @@ spec:
...
               rules:
-                description: Rules represents collection of Rule to apply
+                description: Represents the the array of Oathkeeper access rules to be applied.
                 items:
-                  description: Rule .
+                  description: Rule . 
                   properties:
                     accessStrategies:
-                      description: Set of access strategies for a single path
+                      description: Specifies the list of access strategies.
```
Comment (strekm) [inline installation/resources/crds/api-gateway/apirules.gateway.crd.yaml:62]: can we have here link to oathkeeper docu?

File: installation/resources/crds/api-gateway/apirules.gateway.crd.yaml [hunk 2]
```diff
@@ -47,26 +47,27 @@ spec:
           metadata:
             type: object
           spec:
-            description: APIRuleSpec defines the desired state of ApiRule
+            description: Defines the desired state of the ApiRule.
             properties:
               gateway:
-                description: Gateway to be used
+                description: 	Specifies the Istio Gateway to be used.
```
Comment (strekm) [inline installation/resources/crds/api-gateway/apirules.gateway.crd.yaml:53]: formatting error?

File: installation/resources/crds/api-gateway/apirules.gateway.crd.yaml [hunk 3]
```diff
@@ -82,40 +83,40 @@ spec:
...
-                            description: Name is the name of a handler
+                            description: Specifies the name of the handler.
                             type: string
                         required:
                         - handler
                         type: object
                       type: array
                     path:
-                      description: Path to be exposed
+                      description: 	Specifies the path of the exposed service.
```
Comment (strekm) [inline installation/resources/crds/api-gateway/apirules.gateway.crd.yaml:119]: formatting error?

File: installation/resources/crds/api-gateway/apirules.gateway.crd.yaml [hunk 4]
```diff
@@ -158,22 +159,22 @@ spec:
...
                   code:
-                    description: StatusCode .
+                    description: StatusCode . 
                     type: string
                   desc:
                     type: string
                 type: object
               accessRuleStatus:
-                description: APIRuleResourceStatus .
+                description: APIRuleResourceStatus .
```
Comment (strekm) [inline installation/resources/crds/api-gateway/apirules.gateway.crd.yaml:174]: Describes status of ORY Oathkeeper Rule

File: installation/resources/crds/api-gateway/apirules.gateway.crd.yaml [hunk 5]
```diff
@@ -185,10 +186,10 @@ spec:
                 format: int64
                 type: integer
               virtualServiceStatus:
-                description: APIRuleResourceStatus .
+                description: APIRuleResourceStatus .
```
Comment (strekm) [inline installation/resources/crds/api-gateway/apirules.gateway.crd.yaml:189]: Describes status of Istio VirtualService

File: installation/resources/crds/api-gateway/apirules.gateway.crd.yaml [hunk 6]
```diff
@@ -158,22 +159,22 @@ spec:
...
                   desc:
                     type: string
                 type: object
               accessRuleStatus:
-                description: APIRuleResourceStatus .
+                description: APIRuleResourceStatus . 
                 properties:
                   code:
-                    description: StatusCode .
+                    description: StatusCode .
```
Comment (strekm) [inline installation/resources/crds/api-gateway/apirules.gateway.crd.yaml:177]: Status code describing the ORY Oathkeeper Rule.

File: installation/resources/crds/api-gateway/apirules.gateway.crd.yaml [hunk 7]
```diff
@@ -185,10 +186,10 @@ spec:
                 format: int64
                 type: integer
               virtualServiceStatus:
-                description: APIRuleResourceStatus .
+                description: APIRuleResourceStatus . 
                 properties:
                   code:
-                    description: StatusCode .
+                    description: StatusCode .
```
Comment (strekm) [inline installation/resources/crds/api-gateway/apirules.gateway.crd.yaml:192]: Status code describing the Istio VirtualService.

File: installation/resources/crds/api-gateway/apirules.gateway.crd.yaml [hunk 8]
```diff
@@ -158,22 +159,22 @@ spec:
...
-            description: APIRuleStatus defines the observed state of ApiRule
+            description: Describes the observed state of ApiRule.
             properties:
               APIRuleStatus:
-                description: APIRuleResourceStatus .
+                description: APIRuleResourceStatus . 
                 properties:
                   code:
-                    description: StatusCode .
+                    description: StatusCode .
```
Comment (strekm) [inline installation/resources/crds/api-gateway/apirules.gateway.crd.yaml:168]: Status code describing the APIRule.

**Review by barchw — CHANGES_REQUESTED** [review]

File: installation/resources/crds/api-gateway/apirules.gateway.crd.yaml (hunk 2 above)
Comment (nataliasitko) [reply installation/resources/crds/api-gateway/apirules.gateway.crd.yaml]: fixed

File: docs/05-technical-reference/00-custom-resources/apix-01-apirule.md [hunk 9]
```diff
@@ -74,39 +74,106 @@ spec:
...
+### APIRule.gateway.kyma-project.io/v1beta1
+
+**Spec:**
+
+| Parameter | Type | Description |
+| ---- | ----------- | ---- |
+| **gateway** (required) | string | Specifies the Istio Gateway to be used. |
+| **host** (required) | string | Specifies the URL of the exposed service. |
+| **rules** (required) | \[\]object | Represents the the array of Oathkeeper access rules to be applied. |
+| **rules.&#x200b;accessStrategies** (required) | \[\]object | Specifies the list of access strategies. The supported access strategies are [Oathkeeper](https://www.ory.sh/docs/oathkeeper/pipeline/authn) `oauth2_introspection`, `jwt`, `noop`, and `allow`. |
```
Comment (barchw) [inline docs/05-technical-reference/00-custom-resources/apix-01-apirule.md:91]: The supported ones are all of those listed in https://www.ory.sh/docs/oathkeeper/pipeline/authn

**Review by barchw — CHANGES_REQUESTED** [review]

File: docs/05-technical-reference/00-custom-resources/apix-01-apirule.md [hunk 10]
```diff
@@ -74,39 +74,106 @@ spec:
...
+| **rules.&#x200b;accessStrategies.&#x200b;config**  | object | Configures the handler. Configuration keys vary per handler. |
+| **rules.&#x200b;accessStrategies.&#x200b;config.&#x200b;jwks_urls**  | \[\]string | . |
```
Comment (barchw) [inline docs/05-technical-reference/00-custom-resources/apix-01-apirule.md:93]:
```suggestion
| **rules.&#x200b;accessStrategies.&#x200b;config.&#x200b;jwks_urls**  | \[\]string | Array of URLs from which Ory Oathkeeper can retrieve JSON Web Keys for validating JSON Web Token. |
```

File: docs/05-technical-reference/00-custom-resources/apix-01-apirule.md [hunk 11]
```diff
@@ -74,39 +74,106 @@ spec:
...
+| **rules.&#x200b;accessStrategies.&#x200b;config.&#x200b;trusted_issuers**  | \[\]string | . |
```
Comment (barchw) [inline docs/05-technical-reference/00-custom-resources/apix-01-apirule.md:94]:
```suggestion
| **rules.&#x200b;accessStrategies.&#x200b;config.&#x200b;trusted_issuers**  | \[\]string | If `trusted_issuers` is set, the JWT must contain a value for claim `iss` that matches exactly (case-sensitive) one of the values of `trusted_issuers` |
```

File: docs/05-technical-reference/00-custom-resources/apix-01-apirule.md [hunk 12]
```diff
@@ -74,39 +74,106 @@ spec:
...
+| Parameter | Type | Description |
+| ---- | ----------- | ---- |
+| **APIRuleStatus**  | object | Describes the status of APIRule. |
+| **APIRuleStatus.&#x200b;code**  | string | Status code describing APIRule. |
+| **APIRuleStatus.&#x200b;desc**  | string | Explains the status of APIRule. |
+| **accessRuleStatus**  | object | Describes the status of ORY Oathkeeper Rule. |
+| **accessRuleStatus.&#x200b;code**  | string | Status code describing ORY Oathkeeper Rule. |
+| **accessRuleStatus.&#x200b;desc**  | string | Explains the status of ORY Oathkeeper Rule. |
+| **authorizationPolicyStatus**  | object | APIRuleResourceStatus . |
+| **authorizationPolicyStatus.&#x200b;code**  | string | StatusCode . |
```
Comment (barchw) [inline docs/05-technical-reference/00-custom-resources/apix-01-apirule.md:123]:
```suggestion
| **authorizationPolicyStatus.&#x200b;code**  | string | Status code describing Istio Authorization Policy sub resource. |
```

File: docs/05-technical-reference/00-custom-resources/apix-01-apirule.md [hunk 13]
```diff
@@ -74,39 +74,106 @@ spec:
...
+
+| Parameter | Type | Description |
+| ---- | ----------- | ---- |
+| **APIRuleStatus**  | object | Describes the status of APIRule. |
+| **APIRuleStatus.&#x200b;code**  | string | Status code describing APIRule. |
+| **APIRuleStatus.&#x200b;desc**  | string | Explains the status of APIRule. |
+| **accessRuleStatus**  | object | Describes the status of ORY Oathkeeper Rule. |
+| **accessRuleStatus.&#x200b;code**  | string | Status code describing ORY Oathkeeper Rule. |
+| **accessRuleStatus.&#x200b;desc**  | string | Explains the status of ORY Oathkeeper Rule. |
+| **authorizationPolicyStatus**  | object | APIRuleResourceStatus . |
```
Comment (barchw) [inline docs/05-technical-reference/00-custom-resources/apix-01-apirule.md:122]:
```suggestion
| **authorizationPolicyStatus**  | object | Describes the status of Istio Authorization Policy sub resource. |
```

File: docs/05-technical-reference/00-custom-resources/apix-01-apirule.md [hunk 14]
```diff
@@ -74,39 +74,106 @@ spec:
...
+| **authorizationPolicyStatus.&#x200b;code**  | string | StatusCode . |
+| **authorizationPolicyStatus.&#x200b;desc**  | string | . |
```
Comment (barchw) [inline docs/05-technical-reference/00-custom-resources/apix-01-apirule.md:124]:
```suggestion
| **authorizationPolicyStatus.&#x200b;desc**  | string | Explains the status of Istio Authorization Policy sub resource. |
```

File: docs/05-technical-reference/00-custom-resources/apix-01-apirule.md [hunk 15]
```diff
@@ -74,39 +74,106 @@ spec:
...
+| **lastProcessedTime**  | string | . |
```
Comment (barchw) [inline docs/05-technical-reference/00-custom-resources/apix-01-apirule.md:125]:
```suggestion
| **lastProcessedTime**  | string | Timestamp for when the APIRule was last processed by API Gateway controller. |
```

File: docs/05-technical-reference/00-custom-resources/apix-01-apirule.md [hunk 16]
```diff
@@ -74,39 +74,106 @@ spec:
...
+| **observedGeneration**  | integer | . |
```
Comment (barchw) [inline docs/05-technical-reference/00-custom-resources/apix-01-apirule.md:126]:
```suggestion
| **observedGeneration**  | integer | The generation of the resource that was observed by API Gateway controller. |
```

File: docs/05-technical-reference/00-custom-resources/apix-01-apirule.md [hunk 17]
```diff
@@ -74,39 +74,106 @@ spec:
...
+| **requestAuthenticationStatus**  | object | APIRuleResourceStatus . |
```
Comment (barchw) [inline docs/05-technical-reference/00-custom-resources/apix-01-apirule.md:127]:
```suggestion
| **requestAuthenticationStatus**  | object | Describes the status of Istio Request Authentication sub resource. |
```

File: docs/05-technical-reference/00-custom-resources/apix-01-apirule.md [hunk 18]
```diff
@@ -74,39 +74,106 @@ spec:
...
+| **requestAuthenticationStatus.&#x200b;code**  | string | StatusCode . |
```
Comment (barchw) [inline docs/05-technical-reference/00-custom-resources/apix-01-apirule.md:128]:
```suggestion
| **requestAuthenticationStatus.&#x200b;code**  | string | Status code describing the state of Istio Authorization Policy sub resource . |
```

File: docs/05-technical-reference/00-custom-resources/apix-01-apirule.md [hunk 19]
```diff
@@ -74,39 +74,106 @@ spec:
...
+| **requestAuthenticationStatus.&#x200b;desc**  | string | . |
```
Comment (barchw) [inline docs/05-technical-reference/00-custom-resources/apix-01-apirule.md:129]:
```suggestion
| **requestAuthenticationStatus.&#x200b;desc**  | string | Explains the status of Istio Request Authentication sub resource. |
```

File: docs/05-technical-reference/00-custom-resources/apix-01-apirule.md [hunk 20]
```diff
@@ -74,39 +74,106 @@ spec:
...
+
+**Spec:**
+
+| Parameter | Type | Description |
+| ---- | ----------- | ---- |
+| **gateway** (required) | string | Specifies the Istio Gateway to be used. |
+| **rules** (required) | \[\]object | Represents the array of Oathkeeper access rules to be applied. |
+| **rules.&#x200b;accessStrategies** (required) | \[\]object | Specifies the list of access strategies. The supported access strategies are [Oathkeeper](https://www.ory.sh/docs/oathkeeper/pipeline/authn) `oauth2_introspection`, `jwt`, `noop`, and `allow`. |
+| **rules.&#x200b;accessStrategies.&#x200b;config**  | object | Configures the handler. Configuration keys vary per handler. |
+| **rules.&#x200b;accessStrategies.&#x200b;config.&#x200b;jwks_urls**  | \[\]string | . |
```
Comment (barchw) [inline docs/05-technical-reference/00-custom-resources/apix-01-apirule.md:146]:
```suggestion
| **rules.&#x200b;accessStrategies.&#x200b;config.&#x200b;jwks_urls**  | \[\]string | Array of URLs from which Ory Oathkeeper can retrieve JSON Web Keys for validating JSON Web Token. |
```

File: docs/05-technical-reference/00-custom-resources/apix-01-apirule.md [hunk 21]
```diff
@@ -74,39 +74,106 @@ spec:
...
+| **rules.&#x200b;accessStrategies.&#x200b;config.&#x200b;trusted_issuers**  | \[\]string | . |
```
Comment (barchw) [inline docs/05-technical-reference/00-custom-resources/apix-01-apirule.md:147]:
```suggestion
| **rules.&#x200b;accessStrategies.&#x200b;config.&#x200b;trusted_issuers**  | \[\]string | If `trusted_issuers` is set, the JWT must contain a value for claim `iss` that matches exactly (case-sensitive) one of the values of `trusted_issuers`. |
```

File: docs/05-technical-reference/00-custom-resources/apix-01-apirule.md [hunk 22]
```diff
@@ -74,39 +74,106 @@ spec:
...
+| Parameter | Type | Description |
+| ---- | ----------- | ---- |
+| **APIRuleStatus**  | object | Describes the status of APIRule. |
+| **APIRuleStatus.&#x200b;code**  | string | Status code describing APIRule. |
+| **APIRuleStatus.&#x200b;desc**  | string | Explains the status of APIRule. |
+| **accessRuleStatus**  | object | Describes the status of ORY Oathkeeper Rule. |
+| **accessRuleStatus.&#x200b;code**  | string | Status code describing ORY Oathkeeper Rule. |
+| **accessRuleStatus.&#x200b;desc**  | string | Explains the status of ORY Oathkeeper Rule. |
+| **lastProcessedTime**  | string | . |
+| **observedGeneration**  | integer | . |
```
Comment (barchw) [inline docs/05-technical-reference/00-custom-resources/apix-01-apirule.md:171]:
```suggestion
| **observedGeneration**  | integer | The generation of the resource that was observed by API Gateway controller. |
```

File: docs/05-technical-reference/00-custom-resources/apix-01-apirule.md [hunk 23]
```diff
@@ -74,39 +74,106 @@ spec:
...
+
+| Parameter | Type | Description |
+| ---- | ----------- | ---- |
+| **APIRuleStatus**  | object | Describes the status of APIRule. |
+| **APIRuleStatus.&#x200b;code**  | string | Status code describing APIRule. |
+| **APIRuleStatus.&#x200b;desc**  | string | Explains the status of APIRule. |
+| **accessRuleStatus**  | object | Describes the status of ORY Oathkeeper Rule. |
+| **accessRuleStatus.&#x200b;code**  | string | Status code describing ORY Oathkeeper Rule. |
+| **accessRuleStatus.&#x200b;desc**  | string | Explains the status of ORY Oathkeeper Rule. |
+| **lastProcessedTime**  | string | . |
```
Comment (barchw) [inline docs/05-technical-reference/00-custom-resources/apix-01-apirule.md:170]:
```suggestion
| **lastProcessedTime**  | string | Timestamp for when the APIRule was last processed by API Gateway controller. |
```

**Review by barchw — CHANGES_REQUESTED** [review]

**Review by barchw — DISMISSED** [review]

**Review by barchw — APPROVED** [review]
/woof

**Review by pbochynski — APPROVED** [review]

**Review by IwonaLanger — APPROVED** [review]

**Pull request merged**